http/handler.go

```diff
@@ -303,6 +304,21 @@ func wrapGenericHandler(core *vault.Core, h http.Handler, props *vault.HandlerPr
 		r = r.WithContext(ctx)
 		r = r.WithContext(namespace.ContextWithNamespace(r.Context(), namespace.RootNamespace))
 
+		// Set some response headers with raft node id (if applicable) and hostname, if available
+		nodeID := core.GetRaftNodeID()
+		if nodeID != "" {
+			w.Header().Set("X-Vault-Raft-Node-ID", nodeID)
+		}
+		info, err := hostutil.CollectHostInfo(ctx)
+
+		// These headers aren't essential, so don't return an error here - just log it
+		if err != nil {
+			core.Logger().Error("error retrieving host info", "err", err)
```

Comment:
I don't think we should log this for every request.  I'm not sure any logging is called for really; maybe put a warning in the response instead?

Reply:
It's not being logged on every request - only if there's an error retrieving the host information, which I'm guessing will be relatively rare. My thought was that if customers are relying on this information (per the impetus for this PR) and the header is missing, they're probably going to wonder why it's missing and look at the logs.

Reply:
My thinking is that if it fails, it's likely to fail constantly, in which case upon upgrade suddenly the logs will explode.

Reply:
Can you elaborate more on your suggestion to put a warning in the response? I'm guessing we're not wanting to modify HTTP response bodies because of an error retrieving the hostname, but I'm probably not understanding your suggestion.

If we don't want to log this, my next thought is if it errors, maybe we just ignore the error, omit the header, and document that behavior? I browsed the source for the implementation of `os.Hostname()` and found that on a few platforms, it's using the `uname` system call, a few others use the `hostname` system call. I'm not an expert on system calls, but I'm guessing these are likely fairly stable.